src/phoenix/server/api/mutations/dataset_mutations.py

```diff
@@ -300,6 +305,112 @@ async def delete_dataset(
+        version_description = input.version_description or None
+        version_metadata = input.version_metadata or {}
+        async with info.context.db() as session:
+            datasets = (
+                await session.scalars(
+                    select(models.Dataset)
+                    .join(
+                        models.DatasetExample,
+                        onclause=models.Dataset.id == models.DatasetExample.dataset_id,
+                    )
+                    .where(
+                        models.DatasetExample.id.in_(example_ids),
+                    )
+                )
+            ).all()
+            if not datasets:
+                raise ValueError("No examples found.")
+            if len(set(ds.id for ds in datasets)) > 1:
+                raise ValueError("Examples must come from the same dataset.")
+            dataset = datasets[0]
+
+            previous_revision_ids = (
+                select(func.max(models.DatasetExampleRevision.id))
+                .where(models.DatasetExampleRevision.dataset_example_id.in_(example_ids))
+                .group_by(models.DatasetExampleRevision.dataset_example_id)
+                .scalar_subquery()
+            )
+            previous_revisions = (
+                await session.scalars(
+                    select(models.DatasetExampleRevision)
+                    .where(
+                        and_(
+                            models.DatasetExampleRevision.id.in_(previous_revision_ids),
+                            models.DatasetExampleRevision.revision_kind != "DELETE",
+                        )
+                    )
+                    .order_by(
+                        case(
+                            *(
+                                (
+                                    models.DatasetExampleRevision.dataset_example_id == example_id,
+                                    index,
+                                )
+                                for index, example_id in enumerate(example_ids)
+                            ),
+                            else_=None,
+                        )
+                    )  # ensure the order of the revisions matches the order of the input patches
+                )
+            ).all()
+            if (num_missing_examples := len(example_ids) - len(previous_revisions)) > 0:
+                raise ValueError(f"{num_missing_examples} example(s) could not be found.")
+
+            version_id = await session.scalar(
+                insert(models.DatasetVersion)
+                .returning(models.DatasetVersion.id)
+                .values(
+                    dataset_id=dataset.id,
+                    description=version_description,
+                    metadata_=version_metadata,
+                )
+            )
+            assert version_id is not None
+
+            await session.execute(
+                insert(models.DatasetExampleRevision),
+                [
+                    _to_orm_revision(
```

Comment:
Perhaps we could also set the same `created_at` value as the `DatasetVersion` record?
```suggestion
                    _to_orm_revision(
                        created_at=created_at,

Reply:
If we go that route, I think we probably want to remove the `created_at` field from `dataset_example_revisions` entirely. It would require another join, but probably not the end of the world since we are typically doing subqueries on the versions table to check whether a version exists.